src/main/java/com/google/sps/servlets/AnswerQuestionServlet.java

```diff
@@ -0,0 +1,156 @@
+package com.google.sps.servlets;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.QuerySnapshot;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseAuthException;
+import com.google.firebase.auth.FirebaseToken;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.sps.data.GameInstance;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/answer")
+public class AnswerQuestionServlet extends HttpServlet {
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    JsonObject jsonObj = new Gson().fromJson(request.getReader(), JsonObject.class);
+
+    String userId = getUserId(jsonObj.get("idToken").getAsString());
+
+    String gameInstanceId = jsonObj.get("gameInstanceId").getAsString();
+
+    String gameId = jsonObj.get("gameId").getAsString();
+
+    String questionId = jsonObj.get("questionId").getAsString();
+
+    String answerId = jsonObj.get("answerId").getAsString();
+
+    Firestore firestoreDb = (Firestore) this.getServletContext().getAttribute("firestoreDb");
+
+    DocumentReference gameInstanceDocRef = firestoreDb.collection("gameInstance").document(gameInstanceId);
+
+    boolean isQuestionActive = isQuestionActive(questionId, gameInstanceDocRef);
+
+    if (!isQuestionActive) {
+      return;
+    }
+
+    DocumentReference questionDocRef = firestoreDb.collection("games").document(gameId).collection("questions")
+        .document(questionId);
+
+    String correctAnswerId = getCorrectAnswer(questionId, questionDocRef);
+
+    if (answerId.equals(correctAnswerId)) {
+      System.out.println("The answer is correct");
+
```

Comment:
I'm gonna add the json object return here :)

Reply:
Okay!